components/admin/lessons/AdminLessonInputs.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useState } from 'react'
+import { useAddModuleMutation } from '../../../graphql'
+import { formChange } from '../../../helpers/formChange'
+import { FormCard, Option } from '../../FormCard'
+import { AlertFillIcon, CheckCircleIcon } from '@primer/octicons-react'
+import styles from '../../../scss/adminLessonInputs.module.scss'
+import { Spinner } from 'react-bootstrap'
+import { GraphQLError } from 'graphql-request/dist/types'
+
+type Props = {
+  lesson: { title: string; id: number }
+}
+
+enum Error {
+  INVALID_DATA = 'missing module name or description'
+}
+
+const values: Option[] = [
+  {
+    title: 'Module Name'
+  },
+  {
+    title: 'Description',
+    type: 'MD_INPUT'
+  }
+]
+
+const AdminLessonInputs = ({ lesson }: Props) => {
+  const [formOptions, setFormOptions] = useState(values)
```

Comment:
Can you define a type for `formOptions`? That should remove the need to do type casting on line 34 and 36. Type casting basically tells typescript `stop type checking and trust me, I know what I’m doing`, which weakens type safety. So optimally we should not use type casting unless it is the most appropriate choice

Reply:
Is this resolved?

Reply:
yeah just follow up with code changes as per our discussion about avoiding using type casting